File: algorithms/bubble_sort/python/bubblesort.py
def bubble_sort(A):
    for i in range(0, len(A) - 1):
        for j in range(len(A) - 1, i, -1):
            if A[j] < A[j - 1]:
                aux = A[j]
                A[j] = A[j - 1]
                A[j - 1] = aux
            

File: algorithms/bubble_sort/ruby/bubble_sort.rb
def bubble_sort(array)
  n = array.length
  loop do
    swapped = false

    (n-1).times do |i|
      next unless array[i] > array[i+1]
      array[i], array[i+1] = array[i+1], array[i]
      swapped = true
    end

    break if not swapped
  end

  array
end

arr = [1,4,5,2,6,7,8,3]
bubble_sort(arr)
# => [1, 2, 3, 4, 5, 6, 7, 8]
